core/shared/src/main/scala/zio/FiberId.scala

```diff
@@ -85,12 +85,48 @@ object FiberId {
   def apply(id: Int, startTimeSeconds: Int, location: Trace): FiberId =
     Runtime(id, startTimeSeconds * 1000L, location)
 
-  private[zio] def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime = {
-    val id = ThreadLocalRandom.current().nextInt(Int.MaxValue)
-    FiberId.Runtime(id, java.lang.System.currentTimeMillis(), location)
-  }
+  @deprecated("use `generate` instead", "1.0.0")
+  private[zio] def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime =
+    Gen.Live.make(location)
+
+  private[zio] def generate(fiberRefs: FiberRefs)(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime =
+    fiberRefs.getOrDefault(FiberRef.currentFiberIdGenerator).make(location)
 
   case object None                                                          extends FiberId
   final case class Runtime(id: Int, startTimeMillis: Long, location: Trace) extends FiberId
   final case class Composite(left: FiberId, right: FiberId)                 extends FiberId
+
+  private[zio] trait Gen {
+    def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime
+  }
+
+  private[zio] object Gen {
+
+    /**
+     * Generates a fiber ID where the `id` is a random integer.
+     *
+     * This is more performant than using `FiberId.Gen.Ordered`, but cannot be
+     * used in cases that rely on strict ordering of fibers (e.g., in zio-test)
+     */
+    object Live extends Gen {
+      def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime = {
```

Comment:
@ghostdogpr , isn't it possible to use something like the current thread's id instead of relying on random?
the way I see it we want (mostly) time ordered fiber Ids with a tie breaker, I suspect the spawning thread ID is a good enough tiebreaker and if not it we can add a thread local counter

Reply:
We can't use the thread id here because if the same thread is spawning multiple fibers (e.g., when using `foreachPar`), then the `FiberId.Runtime` will be the same for all the child fibers within the same millisecond, which is bad as IDs will no longer be unique.

As for using a thread local counter, unfortunately that wouldn't work with Loom very well since we get a new Thread per task. There's also no guarantee that 2 separate fibers would need to create a child fiber at the same time and end up having the same value in the counter. Which means we'd have to include both the parent thread ID and a thread local counter in FiberId.Runtime, which breaks both binary and source compatibility :(

Reply:
Also, fiber ordering is only needed for `zio-test`, so for normal execution we don't need ordering at all